dash/testing/wait.py

```diff
@@ -0,0 +1,39 @@
+"""Utils methods for pytest-dash such wait_for wrappers"""
+import time
+from dash.exceptions import TestingTimeoutError
+
+
+def until(
+    wait_cond,
+    timeout,
+    poll=0.1,
+    msg="expected condition not met within timeout",
+):  # noqa: C0330
+    end_time = time.time() + timeout
+    while wait_cond():
+        time.sleep(poll)
+        if time.time() > end_time:
+            raise TestingTimeoutError(msg)
+
+
+def until_not(
+    wait_cond, timeout, poll=0.1, msg="expected condition met within timeout"
+):  # noqa: C0330
+    end_time = time.time() + timeout
+    while not wait_cond():
+        time.sleep(poll)
+        if time.time() > end_time:
+            raise TestingTimeoutError(msg)
+
+
+class text_to_equal(object):  # pylint: disable=too-few-public-methods
```

Comment:
nbd, but could be a higher-order function and the linter wouldn't complain :)
```py
def text_to_equal(selector, text)
    def _call(driver):
        ...

    return _call
```

Reply:
this is actually a mimic from python EC conditions implementation. even though I feel that part is too-java, but just follow the pattern to make it more consistent.